src/rgw/rgw_rest_user_policy.cc

```diff
@@ -142,15 +143,32 @@ void RGWPutUserPolicy::execute(optional_yield y)
 
   try {
     const Policy p(s->cct, s->user->get_tenant(), bl);
+    uint32_t pol_size_occupied = user->get_user_policy_size();
+    uint32_t pol_size = policy.size();
+
     map<string, string> policies;
     if (auto it = user->get_attrs().find(RGW_ATTR_USER_POLICY); it != user->get_attrs().end()) {
       bufferlist out_bl = it->second;
       decode(policies, out_bl);
     }
+
+    //if it is an update
+    if (auto it = policies.find(policy_name); it != policies.end()) {
+      pol_size_occupied = pol_size_occupied - (it->second).size();
+    }
```

Comment:
```suggestion
    if (policies.find(policy_name) != policies.end())
      pol_size_occupied -= (it->second).size();
```
1. tyring to reduce the number of variables
2. we can avoid using `a = a + b` where ever possible, instead use `a += b`. It helps skip one copy operation.

Reply:
done

Reply:
if (policies.find(policy_name) != policies.end())
      pol_size_occupied -= (it->second).size();   this case we cannot access  it->second

but a += b we can do

Reply:
done